CliClient/tests/feature_ThemeAutoDetect.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable no-unused-vars */
+require('app-module-path').addPath(__dirname);
+const { setupDatabaseAndSynchronizer, switchClient, asyncTest, TestApp } = require('test-utils.js');
+const Setting = require('lib/models/Setting.js');
+const { time } = require('lib/time-utils.js');
+
+let testApp = null;
+
+describe('integration_ThemeAutoDetect', function() {
+
+	beforeEach(async (done) => {
+		testApp = new TestApp();
+		await testApp.start(['--no-welcome']);
+		done();
+	});
+
+	afterEach(async (done) => {
+		if (testApp !== null) await testApp.destroy();
+		testApp = null;
+		done();
+	});
+
+	it('should update value of preferred light theme setting', asyncTest(async () => {
+		// SETUP: Set Preferred Light Theme option to Solarized Light Theme value
+		testApp.dispatch({
+			type: 'SETTING_UPDATE_ONE',
+			key: 'preferredLightTheme',
+			value: Setting.THEME_SOLARIZED_LIGHT,
+		});
+		await testApp.wait();
+
+		const state = testApp.store().getState();
+		// changing preferredLightTheme option should change theme value
+		expect(state.settings.preferredLightTheme).toBe(Setting.THEME_SOLARIZED_LIGHT);
+	}));
+
+	it('should update value of preferred dark theme setting', asyncTest(async () => {
+		// SETUP: Set Preferred Dark Theme option to Dark Theme value
+		testApp.dispatch({
+			type: 'SETTING_UPDATE_ONE',
+			key: 'preferredDarkTheme',
+			value: Setting.THEME_DARK,
+		});
+		await testApp.wait();
+
+		const state = testApp.store().getState();
+		// changing preferredDarkTheme option should change theme value
+		expect(state.settings.preferredDarkTheme).toBe(Setting.THEME_DARK);
+	}));
+
+	it('should update themeAutoDetect checkbox setting', asyncTest(async () => {
+		// SETUP: Set themeAutoDetect checkbox to true
+		testApp.dispatch({
+			type: 'SETTING_UPDATE_ONE',
+			key: 'themeAutoDetect',
+			value: true,
+		});
+		await testApp.wait();
+
+		const state = testApp.store().getState();
+		expect(state.settings.themeAutoDetect).toBe(true);
+	}));
+
+	it('should update Night Mode setting', asyncTest(async () => {
+		// SETUP: Set Night Mode value to true
+		testApp.dispatch({
+			type: 'SETTING_UPDATE_ONE',
+			key: 'isNightMode',
+			value: true,
+		});
+		await testApp.wait();
+
+		const state = testApp.store().getState();
+		expect(state.settings.isNightMode).toBe(true);
+	}));
+
+});
```

Comment:
All the above tests are not necessary. In general if we set a setting value, its value should indeed be changed and that's probably already tested somewhere else in a more generic way.  We don't need to check this for every single setting value. However, there might be tests specific to your feature that might be needed.

Reply:
I can't thought of any other tests, because the changes I have made are specific to Electron Client and the build tests are related to changes in React Native Client.. if you thought of any specific tests then please let me know I'll try.

Reply:
I agree with Laurent, this test is not necessary as settings are tested elsewhere.  Also agree with you that as the changes are local to the Electron client, there is no scope for automatic testing at this time.

Reply:
I'm sorry for the late response I didn't checked your edited message. The changes I have made in settings is just addition of new variables: `preferredDarkTheme`, `preferredLightTheme`, `isNightMode` and for all these variables as Laurent said it's already tested somewhere else in a more generic way so I didn't thought of any need of build tests here